src/components/signIn.js

```diff
@@ -20,13 +24,27 @@ const SignIn = (props) => {
       },
       body: JSON.stringify( {"username": userName, "password": passWord} )
       })
-    .then((response) => response.json())
-    .then((data) => {
-       console.log(data.loggedIn)
-      if (data.loggedIn) {
-        console.log(data)
-        props.navigation.navigate('Parrot List');
+      .then((response) => response.json())
+      .then((data) => {
+        console.log("returned data", data);
+        // data here is: sessionId, userId, username, userType
+        if (data.sessionId) {
+          setUserType(data.userType);
+          setUserId(data.userId);
+          setSessionId(data.sessionId);
+
+          // user Id is not being set!! (probably type and session too)
```

Comment:
Hi! I was looking at this earlier today and if you put the console.log outside this function, then you can see that the userID is being set (and the rest of the useState const). I think it's because useState is not set straight away when you call it cos its asynchronous also and we're just telling react to setState at" some point in the future" and React will then set when its re renders or something. So we can keep using state.